crates/bls-gadgets/src/y_to_bit.rs

```diff
@@ -244,13 +288,218 @@ mod test {
             0x5d, 0xbe, 0x62, 0x59, 0x8d, 0x31, 0x3d, 0x76, 0x32, 0x37, 0xdb, 0x17, 0xe5, 0xbc,
             0x06, 0x54,
         ]);
+
+        // Check random points.
         for i in 0..10 {
             let secret_key = Bls12_377Fr::rand(rng);
 
             let generator = Bls12_377G2Projective::prime_subgroup_generator();
             let pub_key = generator.clone().mul(secret_key);
 
             let half = <Bls12_377Parameters as Bls12Parameters>::Fp::modulus_minus_one_div_two();
+            let zero = <Bls12_377Parameters as Bls12Parameters>::Fp::zero();
+
+            {
+                let mut cs = TestConstraintSystem::<SW6Fr>::new();
+
+                let pk =
+                    G2Gadget::<Bls12_377Parameters>::alloc(&mut cs.ns(|| "alloc"), || Ok(pub_key))
+                        .unwrap();
+
+                let y_bit =
+                    YToBitGadget::<Bls12_377Parameters>::y_to_bit_g2(cs.ns(|| "y to bit"), &pk)
+                        .unwrap();
+
+                if pk.y.c1.get_value().get().unwrap().into_repr() > half
```

Comment:
"if c1 > half or c1 == 0 && c0 > half" -> 1, else 0 - OK this matching https://github.com/celo-org/bls-zexe/issues/149#issue-601795833